pkgs/applications/networking/browsers/vivaldi/default.nix

```diff
@@ -10,23 +10,23 @@
 }:
 
 let
-  archUrl = name: arch: "https://vivaldi.com/download/stable/${name}_${arch}.deb";
+  archUrl = name: arch: "https://downloads.vivaldi.com/stable/${name}_${arch}.deb";
 in
 stdenv.mkDerivation rec {
-  version    = "1.0";
-  debversion = "stable_1.0.435.40-1";
+  version    = "1.1";
+  debversion = "stable_1.1.453.47-1";
   product    = "vivaldi";
   name       = "${product}-${version}";
 
   src = if stdenv.system == "x86_64-linux"
```

Comment:
The idea is to do something like this:
```nix
{
  "x86_64-linux" = {
    sha256 = "...";
    arch = "amd64";
  };

  "i686-linux" = {
    # as above
  };
}."${stdenv.system}" or (throw "unsupported system: ${stdenv.system}"
```

Also note that the only thing that changes for the URL is the `arch` part, so the `archUrl` function is fairly
redundant.